packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -333,7 +333,75 @@ class ShaderMask extends SingleChildRenderObjectWidget {
   }
 }
 
-/// A widget that applies a filter to the existing painted content and then paints [child].
+/// A widget that applies a filter to the existing painted content and then
+/// paints [child].
+///
+/// The filter will only be applied to the area of the background in which the
+/// [child] (or one of its descendants) is actually going to paint in regardless
+/// of the actual size of [child].
+///
+/// In more technical terms, [BackdropFilter] will only apply the filter to the
+/// area of the background that is covered by the [Scene] defined by [child]
+/// and its descendants.
+///
+/// {@tool sample}
+/// Even though the [BackdropFilter] is wrapping the [Container] below, the
+/// background will only be blurred in the area defined by the bounding box
+/// of the [Text] because that's the only area any descendant of the
+/// [BackdropFilter] is painting in.
+///
+/// ```dart
+/// Stack(
+///   fit: StackFit.expand,
+///   children: <Widget>[
+///     Text('0' * 10000),
+///     Center(
+///       child: BackdropFilter(
+///         filter: ui.ImageFilter.blur(
+///           sigmaX: 5.0,
+///           sigmaY: 5.0,
+///         ),
+///         child: Container(
+///           alignment: Alignment.center,
+///           width: 200.0,
+///           height: 200.0,
+///           child: Text('Hello World'),
+///         ),
+///       ),
+///     ),
+///   ],
+/// )
+/// ```
+///
+/// To blur the entire area of the [Container], increase the paint area of
+/// the Container. Giving it a transparent background color will increase
+/// the paint area of the container (and hence blur the background behind the
+/// entire container) without changing other visual properties.
+///
+/// ```dart
+/// Stack(
+///   fit: StackFit.expand,
+///   children: <Widget>[
+///     Text('0' * 10000),
+///     Center(
+///       child: BackdropFilter(
+///         filter: ui.ImageFilter.blur(
+///           sigmaX: 5.0,
+///           sigmaY: 5.0,
+///         ),
+///         child: Container(
+///           color: Colors.transparent, // <-- NEW
+///           alignment: Alignment.center,
+///           width: 200.0,
+///           height: 200.0,
+///           child: Text('Hello World'),
+///         ),
+///       ),
+///     ),
+///   ],
+/// )
+/// ```
+/// {@end-tool}
```

Comment:
this is begging for screenshots.

Reply:
I will add code that generates them to https://github.com/flutter/assets-for-api-docs. For further discussion here's what the two samples produce when running on a phone:

![img_0010](https://user-images.githubusercontent.com/1227763/52886900-ba919000-312a-11e9-91c8-e461d7d845dd.PNG)

![img_0011](https://user-images.githubusercontent.com/1227763/52886908-be251700-312a-11e9-8216-d856f26128be.PNG)

Reply:
Actually, our diagram generating infrastructure currently can't handle blurs: https://github.com/flutter/flutter/issues/28028